src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -588,27 +623,38 @@ protected virtual void OnDeleteBlocks(DBreeze.Transactions.Transaction dbreezeTr
         private List<Block> GetBlocksFromHashes(DBreeze.Transactions.Transaction dbreezeTransaction, List<uint256> hashes)
         {
             this.logger.LogTrace("({0}.{1}:{2})", nameof(hashes), nameof(hashes.Count), hashes?.Count);
+            
+            var results = new Dictionary<uint256, Block>();
 
-            var blocks = new List<Block>();
+            // Access hash keys in sorted order.
+            var byteListComparer = new ByteListComparer();
+            var keys = hashes.Select(hash => (hash, hash.ToBytes())).ToList();
 
-            foreach (uint256 hash in hashes)
-            {
-                byte[] key = hash.ToBytes();
+            keys.Sort((key1, key2) => byteListComparer.Compare(key1.Item2, key2.Item2));
 
-                Row<byte[], Block> blockRow = dbreezeTransaction.Select<byte[], Block>("Block", key);
+            foreach (var key in keys)
+            {
+                Row<byte[], Block> blockRow = dbreezeTransaction.Select<byte[], Block>("Block", key.Item2);
```

Comment:
Here you are selecting each block separately. 
If  you use `SelectForwardFromTo` with from = first item of the sorted keys array and to = last item then selection will be much faster because DBreeze search algorithm won't have to start from the first key in the dictionary each time, with select forward it will continue where it left (but ofc it works only with sorted list of keys but you have one).

Reply:
I am unconvinced it will be faster. Due to the distribution of the keys it will probably not be faster on average case, or at least we should measure that before we go for that optimisation.

Reply:
We are doing an ordered sparse sequential selection from a sorted sequence so its unlikely we will be getting more than one item at a time - i.e. I don't see how `SelectForwardFromTo` will be helpful when mostly returning one item at a time.

Reply:
If I understand SelectForwardFromTo correctly (may not be the case, never used it), it does not return a range, but rather specifies the method of finding the matching items. So in my understanding we should not receive items we have not asked for, so the whole argument seems to me to be only about efficiency. I would still like to see some actual measurements on the fully synced mainnet node data.